examples/server/server.cpp

```diff
@@ -3014,8 +3018,14 @@ int main(int argc, char **argv)
         std::placeholders::_2,
         std::placeholders::_3
     ));
-    llama.queue_tasks.start_loop();
 
+    shutdown_handler = [&](int) {
+        llama.queue_tasks.terminate();
+    };
+    signal(SIGTERM, signal_handler);
+    signal(SIGINT, signal_handler);
```

Comment:
The other examples use sigaction() on Unix (signal() is effectively [deprecated](https://pubs.opengroup.org/onlinepubs/009604499/functions/signal.html#tag_03_690_07) by POSIX) and SetConsoleCtrlHandler on Windows. We should do something similar here.

Reply:
Updated.